Source/WebCore/rendering/RenderBox.cpp

```diff
@@ -176,6 +176,7 @@ void RenderBox::willBeDestroyed()
 
     view().unscheduleLazyRepaint(*this);
     removeControlStatesForRenderer(*this);
+    theme().removeControlPartForRenderer(*this);
```

Comment:
Shame that we place this cleanup burden on RenderBox. Can we do this some other way?

Reply:
Fixe. This line was removed since the renderer `ControlPart` was moved to `RenderObjectRareData`.